### DISCUSSION TITLE
How can I login a user by username or email in my laravel 8 project.

### DISCUSSION THREAD

**Discussion by ShekhSaifuddin007** [laravel/framework#34868] [General]
I'm creating a project with fortify authentication and jetstream, in my project requirement is a user can login with his/her username or email, but in laravel 8 I don't find the logic or any resource on that, how can I achieve this. help is appreciable.

**Comment by sektonic** [comment]
This my solution:
1. Go to config/portify.php and find line 'username' => 'email' change to 'username' => 'username'
2. Then go to app/Providers/JetstreamServiceProvider.php and add the following code:
```
// Add use
use App\Models\User;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Hash;
use Laravel\Fortify\Fortify;

public function boot()
{
    // ...

    // Add this code
    Fortify::authenticateUsing(function (Request $request) {
	$login = $request->username;
	if (filter_var($login, FILTER_VALIDATE_EMAIL)) {
		$user = User::where('email', $login)->first();
	} else {
		$user = User::where('username', $login)->first();
	}

	if ($user && Hash::check($request->password, $user->password)) {
		return $user;
	}
    });
}
```

**Reply by masoudganji** [reply]
It's OK but:
3. In file "app\Http\Controllers\Auth\LoginController.php" add this:
public function username() {
    return 'username';
}

**Reply by ShekhSaifuddin007** [reply]
Still now you are in old school bro, this is fortify.. 😊😊

**Reply by masoudganji** [reply]
I use Laravel 8. The first answer was not my solution and after adding function username() it works.

**Reply by ShekhSaifuddin007** [reply]
Yah first answer have little mistake it not will be in jetstream service provider it will be in fortify service provider

Btw where you write this username function..??

**Reply by masoudganji** [reply]
app\Http\Controllers\Auth\LoginController.php

**Reply by ShekhSaifuddin007** [reply]
Then you are not using fortify for authentication you are using laravel/ui for authentication right..??

**Reply by masoudganji** [reply]
I think so

**Reply by Messhias** [reply]
The user function username on this case probably enters like a magic function request in that case?
